tests/values.go

```diff
@@ -0,0 +1,245 @@
+package tests
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"golang.org/x/xerrors"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/engine"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+var _ = fmt.Stringer(CoderValues{})
+
+// CoderValues is a typed Go representation of Coder's values file,
+// suitable for writing tests. This provides code completion for Go
+// tests.
+type CoderValues struct {
```

Comment:
Keeping this in sync manually is going to be an issue down the line.
Could we generate this automatically from `values.yaml`? 

Aside: _is_ there any decent tool for doing this directly? I found [yaml2go](https://github.com/PrasadG193/yaml2go) but it appears to not be maintained. We could also go the route of YAML -> JSON -> Go, for which there is a [slightly more up to date tool](https://github.com/m-zajac/json2go). If all else fails, I suppose there's Swagger.

Reply:
I discussed this with @Emyrk, there's a way you can define a JSON Schema to validate the values, and we could generate both the values.yaml and these structs from that. But in the interest of an initial implementation, I left taht out for now :)

Reply:
Nice :-) I think this will really help!